PR: Fix/Find in files

goanpeca opened the pull request.

ccordoba12 commented on spyderlib/config/utils.py:
```diff
@@ -80,15 +80,23 @@ def _get_pygments_extensions():
-    return extensions
+            extensions = extensions + list(lexer_exts) + list(other_exts)
+
+    return sorted(list(set(extensions)))
```
What do these changes do exactly?

goanpeca replied:
`lexer_exts = [le[1:] for le in lexer_exts]`

This code is wrong cause not all lexers extensions start with a **`*`**, so I fixed that so the actual generated patterns are valid regex. (this was causing find in files to not work after a clean install)

ccordoba12 commented on spyderlib/plugins/findinfiles.py:
```diff
@@ -94,14 +97,23 @@ def findinfiles_callback(self):
+                    ]
         return patterns
-    
+        
```
Please remove this blank :-)

ccordoba12 commented:
Does your changes hide the default (advanced) view of *Find in files*? If not, I'd like to see that as part of this PR ;-)

goanpeca commented:
Hi @ccordoba12 , I forgot to add the description. Basically the include regex where based on the extension provided by the pygments lexers, but the problem is that the generated regex was invalid. I corrected that and added some tests so we are sure that the Include and Exclude patterns are always valid regular expressions.

I can hide the pane by default. Working on a ack/grin like interface after this PR.

goanpeca commented:
@ccordoba12, now the advanced options are hidden by default

ccordoba12 commented:
I tested our edit filters and they are working correctly, so merging ;-)
